MiniApp/Classes/Display/MiniAppWebView.swift

```diff
@@ -12,21 +12,28 @@ internal class MiniAppWebView: WKWebView {
         return config
     }
 
-    convenience init(miniAppId: String, versionId: String) {
+    convenience init(miniAppId: String, versionId: String, queryParams: String? = nil) {
         let schemeName = Constants.miniAppSchemePrefix + miniAppId
-        let urlRequest = URLRequest(url: URL(string: schemeName + "://miniapp/" + Constants.rootFileName)!)
+        let urlRequest = URLRequest(url: URL(string: schemeName + "://miniapp/" + Constants.rootFileName + MiniAppWebView.getQueryParams(queryParams: queryParams))!)
         let config = MiniAppWebView.defaultConfig()
         config.setURLSchemeHandler(URLSchemeHandler(versionId: versionId), forURLScheme: schemeName)
         self.init(frame: .zero, configuration: config)
         commonInit(urlRequest: urlRequest)
     }
 
-    convenience init(miniAppURL: URL) {
+    convenience init(miniAppURL: URL, queryParams: String? = nil) {
         let urlRequest = URLRequest(url: miniAppURL)
         self.init(frame: .zero, configuration: MiniAppWebView.defaultConfig())
         commonInit(urlRequest: urlRequest)
     }
 
+    private static func getQueryParams(queryParams: String?) -> String {
+        guard let param = queryParams, let queryString =  param.addingPercentEncoding(withAllowedCharacters: .urlQueryAllowed) else {
```

Comment:
I don't think we should use `addingPercentEncoding` - it should be the mini app's responsibility to ensure that they're passing/receiving encoded queries.

Also, they might want to use a URL fragment with `#` like this: `?param1=value#some_other_thing`. I think in this case `urlQueryAllowed` would encode the `#`, right?

Reply:
Ah, yeah. Adding a percent-encoding is required here as the [URL](https://github.com/rakutentech/ios-miniapp/blob/master/MiniApp/Classes/Display/MiniAppWebView.swift#L17) will not accept a string with spaces, it will crash. To form a URL with space, we are adding this encoding.

w/ Encoding
`a b c` **->** `/index.html?a%20b%20c`
`?param1=value#some_other_thing` -> `/index.html??param1=value%23some_other_thing`

w/o encoding
`a b c` **->** CRASH
`?param1=value#some_other_thing` **->** `/index.html??param1=value#some_other_thing`

Reply:
Ah, it's because on iOS you have to load the webview from a URLRequest instead of a plain String?

But it's a problem that the `#` is getting converted to `%23`. Mini apps might want to use a Url fragment. Could you manually parse the `#something` portion first before encoding the query? Looks like there's a separate `urlFragmentAllowed` so you could encode that part separately: https://developer.apple.com/documentation/foundation/nscharacterset/1412537-urlfragmentallowed

Reply:
I found a better way now. Please check